Add Gap Mesh Generator (#32120)
idaholab/moose: modules/reactor/doc/content/source/meshgenerators/GapMeshGenerator.md

# GapMeshGenerator

!syntax description /Mesh/GapMeshGenerator

## Overview

> In the mesh generation for reactor geometries, it is often necessary to create gaps that are conformal to the external boundaries of components. The gap is meant to have a uniform thickness throughout the geometry. This `GapMeshGenerator` class provides functionality to generate a polyline mesh (similar to the meshes generated by [`PolyLineMeshGenerator`](/PolyLineMeshGenerator.md)) with a specified gap thickness based on an 2D input mesh residing in the XY plane. The input mesh must be a connected mesh with only one outer boundary manifold. The generated polyline mesh can then be used for further mesh generation processes, such as [`XYDelaunayGenerator`](/XYDelaunayGenerator.md).

Reviewer: ```suggestion
In the mesh generation for reactor geometries, it is often necessary to create unmeshed gaps along the external boundaries of components. The gap is meant to have a uniform thickness all around the boundary. This `GapMeshGenerator` class provides functionality to generate a polyline mesh (similar to the meshes generated by [`PolyLineMeshGenerator`](/PolyLineMeshGenerator.md)) with a specified gap thickness based on an 2D input mesh residing in the XY plane. The input mesh must be a connected mesh with only one outer boundary manifold. The generated polyline mesh can then be used for further mesh generation processes, such as [`XYDelaunayGenerator`](/XYDelaunayGenerator.md).
```

## Examples

A simple example shown here is based on a set of patterned hexagons. As shown in [simple_gap], the generated polyline mesh without refinement only contains one `EDGE2` element for each side of the hexagons with a gap from the original hexagons.

!media meshgenerators/simple_gap.png
      style=display: block;margin-left:auto;margin-right:auto;width:40%;
      id=simple_gap
      caption=A simple non-refined polyline mesh that is generated based on a set of patterned hexagons with a gap.

After specifying a maximum element size, additional nodes are inserted along each side of the hexagons to create smaller `EDGE2` elements, as shown in [refined_gap]. Finally, using the refined polyline mesh as input, a triangular mesh with a circular outer boundary is generated by [`XYDelaunayGenerator`](/XYDelaunayGenerator.md) as shown in [gap_mesh].

!media meshgenerators/refined_gap.png
      style=display: block;margin-left:auto;margin-right:auto;width:40%;
      id=refined_gap

>       caption=A refined polyline mesh that is generated based on a set of patterned hexagons with a gap.

Reviewer: ```suggestion
      caption=A refined polyline mesh that is generated around a set of patterned hexagons with a gap.
```
